pykafka/producer.py

```diff
@@ -250,3 +250,68 @@ def produce(self, messages):
         # only *some* messages when a leader changes. Therefore, we don't want
         # a random partition distribution changing that on the retry.
         self._produce(self._partition_messages(messages), 0)
+
+
+class AsyncProducer(Producer):
+    """
+    This class implements the asynchronous producer logic found in the
+    JVM driver. In inherits from the synchronous implementation.
+    """
+    def __init__(self, *args, **kwargs):
+        """Instantiate a new AsyncProducer
+
+        For argument documentation, see :class:`pykafka.producer.Producer`
+        """
+        super(AsyncProducer, self).__init__(*args, **kwargs)
+        self._setup_workers()
+
+    def _setup_workers(self):
+        """Spawn workers for connected brokers
+
+        Creates one :class:`queue.Queue` instance and one worker thread per
+        broker. The queue holds requests waiting to be sent, and the worker
+        thread reads the end of the queue and sends requests.
+        """
+        self._queues_by_leader = {}
+        for partition in self._topic.partitions.values():
+            if partition.leader.id not in self._queues_by_leader:
+                self._queues_by_leader[partition.leader.id] = self._setup_queue(partition.leader)
+
+    def _setup_queue(self, broker):
+        """Spawn a worker for the given broker
+
+        Creates a :class:`queue.Queue` instance and a worker thread for the
+        given broker.
+
+        :param broker: The broker for which to instantiate this queue and worker
+        :type broker: :class:`pykafka.broker.Broker`
+        """
+        def worker():
+            while True:
+                time.sleep(.0001)
+                try:
+                    request, attempt = request_queue.get_nowait()
```

Comment:
No, it should block forever and the thread should stay asleep as long as there is no data coming in.  Since it's on a separate thread, is it a problem for it to sit there?